Doc/library/unittest.mock.rst

```diff
@@ -1099,6 +1101,58 @@ object::
       [call('foo'), call('bar')]
 
 
+.. class:: ThreadingMock(spec=None, side_effect=None, return_value=DEFAULT, wraps=None, name=None, spec_set=None, unsafe=False, **kwargs)
+
+  A version of :class:`MagicMock` for multithreading tests. The
+  :class:`ThreadingMock` object provides extra methods to wait for a call to
+  happen on a different thread, rather than assert on it immediately.
+
+  .. method:: block_until_called()
+
+      Blocks until the the mock is called.
+
+        >>> mock = ThreadingMock()
+        >>> thread = threading.Thread(target=mock)
+        >>> thread.start()
+        >>> mock.wait_until_called(1)
+        >>> thread.join()
+
+  .. method:: block_until_any_call(*args, **kwargs)
+
+      Block until the the mock is called with the specified arguments.
+
+        >>> mock = ThreadingMock()
+        >>> thread = threading.Thread(target=mock, args=("arg1", "arg2",), kwargs={"arg": "thing"})
+        >>> thread.start()
+        >>> mock.wait_until_any_call(1, "arg1", "arg2", arg="thing")
```

Comment:
The above then becomes:
```
>>> mock.until_any_call("arg1", "arg2", arg="thing").wait(timeout=1)
```

I'm not sure we actually need the blocking variants, since that could just be spelled:
```
>>> mock.until_any_call("arg1", "arg2", arg="thing").wait()
```
...which I think is the behaviour you proposed (and which I really like!) for the default timeout being `None` and meaning "don't timeout".